ibc_tests/transfer_test.go

```diff
@@ -0,0 +1,170 @@
+	packet = channeltypes.NewPacket(fungibleTokenPacket.GetBytes(), 1, pathBtoC.EndpointA.ChannelConfig.PortID, pathBtoC.EndpointA.ChannelID, pathBtoC.EndpointB.ChannelConfig.PortID, pathBtoC.EndpointB.ChannelID, timeoutHeight, 0)
+	err = pathBtoC.RelayPacket(packet, ack.Acknowledgement())
+	suite.Require().NoError(err) // relay committed
+
+	coinSentFromBToC := sdk.NewCoin(types.ParseDenomTrace(fullDenomPath).IBCDenom(), amount)
+	balance = GetLikeApp(suite.T(), suite.chainC).BankKeeper.GetBalance(suite.chainC.GetContext(), suite.chainC.SenderAccount.GetAddress(), coinSentFromBToC.Denom)
+
+	// check that the balance is updated on chainC
+	suite.Require().Equal(coinSentFromBToC, balance)
+
+	// check that balance on chain B is empty
+	balance = GetLikeApp(suite.T(), suite.chainB).BankKeeper.GetBalance(suite.chainB.GetContext(), suite.chainB.SenderAccount.GetAddress(), coinSentFromBToC.Denom)
+	suite.Require().Zero(balance.Amount.Int64())
+
+	// send from chainC back to chainB
+	msg = types.NewMsgTransfer(pathBtoC.EndpointB.ChannelConfig.PortID, pathBtoC.EndpointB.ChannelID, coinSentFromBToC, suite.chainC.SenderAccount.GetAddress().String(), suite.chainB.SenderAccount.GetAddress().String(), timeoutHeight, 0)
+
+	_, err = suite.chainC.SendMsgs(msg)
+	suite.Require().NoError(err) // message committed
+
+	// relay send
+	// NOTE: fungible token is prefixed with the full trace in order to verify the packet commitment
+	fungibleTokenPacket = types.NewFungibleTokenPacketData(fullDenomPath, coinSentFromBToC.Amount.String(), suite.chainC.SenderAccount.GetAddress().String(), suite.chainB.SenderAccount.GetAddress().String())
+	packet = channeltypes.NewPacket(fungibleTokenPacket.GetBytes(), 1, pathBtoC.EndpointB.ChannelConfig.PortID, pathBtoC.EndpointB.ChannelID, pathBtoC.EndpointA.ChannelConfig.PortID, pathBtoC.EndpointA.ChannelID, timeoutHeight, 0)
+	err = pathBtoC.RelayPacket(packet, ack.Acknowledgement())
+	suite.Require().NoError(err) // relay committed
+
+	balance = GetLikeApp(suite.T(), suite.chainB).BankKeeper.GetBalance(suite.chainB.GetContext(), suite.chainB.SenderAccount.GetAddress(), coinSentFromAToB.Denom)
+
+	// check that the balance on chainA returned back to the original state
+	suite.Require().Equal(coinSentFromAToB, balance)
+
+	// check that module account escrow address is empty
+	escrowAddress := types.GetEscrowAddress(packet.GetDestPort(), packet.GetDestChannel())
+	balance = GetLikeApp(suite.T(), suite.chainB).BankKeeper.GetBalance(suite.chainB.GetContext(), escrowAddress, sdk.DefaultBondDenom)
+	suite.Require().Equal(sdk.NewCoin(sdk.DefaultBondDenom, sdk.ZeroInt()), balance)
+
+	// check that balance on chain B is empty
+	balance = GetLikeApp(suite.T(), suite.chainC).BankKeeper.GetBalance(suite.chainC.GetContext(), suite.chainC.SenderAccount.GetAddress(), voucherDenomTrace.IBCDenom())
+	suite.Require().Zero(balance.Amount.Int64())
```

Comment:
Comment says `chain B` but code in line 164 is getting from `chainC`?

Reply:
I think it meant endpoint B (chain C), let's update the comment